internal/server/middleware.go

```diff
@@ -110,6 +111,25 @@ func authenticateRequest(c *gin.Context, route routeSettings, srv *Server) (acce
 		if err := handleInfraDestinationHeader(rCtx, c.Request.Header); err != nil {
 			return authned, err
 		}
+		if route.idpSync {
+			if route.authenticationOptional {
+				// this should be caught during development
+				return authned, fmt.Errorf("idp sync requires authentication")
+			}
+			// sync the identity info here to keep the UI session in sync with IDP session validity
+			if err := srv.syncIdentityInfo(context.Background(), tx, authned.User, authned.AccessKey.ProviderID); err != nil {
+				deleteCookie(c.Writer, cookieAuthorizationName, c.Request.Host)
+				if errors.Is(err, ErrSyncFailed) {
+					logging.L.Debug().Err(err)
+				} else {
+					logging.L.Error().Err(err)
+				}
+				if err = tx.Commit(); err != nil {
+					logging.L.Error().Err(err)
+				}
```

Comment:
Do we need to commit here on all errors?  I think maybe `syncIdentityInfo` needs to manage its own transaction.